### PR TITLE
Rearrange magic methods to the top

### PR DESCRIPTION
**Pull request by harshil21**
Simply rearranges the magic methods to be at the top of a class since they define class behaviour.

Ref: https://github.com/python-telegram-bot/python-telegram-bot/pull/3826#issuecomment-1666593803

### CODE REVIEW

**Comment by Poolitzer** [comment]
I mean... If you rearrange them anyway... maybe we want them in alphabetical order? :D

**Comment by harshil21** [comment]
Yeah I thought of it but it's too much maintainence work for little benefit imo

File: telegram/_bot.py [hunk 1]
```diff
@@ -308,6 +308,54 @@ def __init__(
 
         self._freeze()
 
+    async def __aenter__(self: BT) -> BT:
+        try:
```
Comment (Bibo-Joshi) [inline telegram/_bot.py:312]: Having just written https://github.com/python-telegram-bot/python-telegram-bot/pull/3826#issuecomment-1714435152 I see that these methods do not have a docstring - although they are implicitly documented in the class docstring. If we decide that they should get a docstring, we can do that in a separate PR, just wanted your opinion on it :)

**Review by Bibo-Joshi — APPROVED** [review]
LGTM - just one related question and conflict resolving is needed :)

File: telegram/_bot.py (hunk 1 above)
Comment (harshil21) [reply telegram/_bot.py:312]: yeah I think they should get a separate docstring, makes it more consistent.

Comment (Bibo-Joshi) [reply telegram/_bot.py:312]: https://github.com/python-telegram-bot/python-telegram-bot/issues/3886
